portfolio/src/main/java/com/google/sps/data/ArrayListTimeRangeGroup.java

```diff
@@ -0,0 +1,187 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+
+/** Models an implementation of the TimeRangeGroup model using ArrayList. */
+public class ArrayListTimeRangeGroup implements TimeRangeGroup {
+
+  // This list of all the time ranges will be sorted by start time ascending
+  // in add and delete methods. The time ranges stored in this list are
+  // pair-wise disjoint at any moment.
+  public List<TimeRange> allTimeRanges;
+
+  /**
+   * Adds all the input time ranges to the list of all time ranges. Also sorts the list of all
+   * ranges in the constructor.
+   */
+  public ArrayListTimeRangeGroup(Collection<TimeRange> timeRanges) {
```

Comment:
I think you might be able to take in Iterable<TimeRange> that is more general than Collection

Reply:
Updated.